pkg/cosmos/client/gas_price_estimator.go

```diff
@@ -17,16 +19,77 @@ var _ GasPricesEstimator = (*FixedGasPriceEstimator)(nil)
 
 type FixedGasPriceEstimator struct {
 	gasPrices map[string]sdk.DecCoin
+	lggr      logger.SugaredLogger
 }
 
-func NewFixedGasPriceEstimator(prices map[string]sdk.DecCoin) *FixedGasPriceEstimator {
-	return &FixedGasPriceEstimator{gasPrices: prices}
+func NewFixedGasPriceEstimator(prices map[string]sdk.DecCoin, lggr logger.SugaredLogger) *FixedGasPriceEstimator {
+	return &FixedGasPriceEstimator{gasPrices: prices, lggr: lggr}
 }
 
 func (gpe *FixedGasPriceEstimator) GasPrices() (map[string]sdk.DecCoin, error) {
 	return gpe.gasPrices, nil
 }
 
+func (gpe *FixedGasPriceEstimator) GasPrice(coin string) sdk.DecCoin {
```

Comment:
Thanks for bringing this up @cfal ! Agreed with your point about how 
1. This function should be more consistent and return `(map, error)` 

2. 
> do we need this new function? vs the caller doing GasPrices()[coin] themselves
Initially, I thought that the getter and setter method would be useful especially when multiple `goroutines` bumps up, and therefore we might need mutexes. However upon deeper analysis of `TxnAttemptBuilder` and confirmer logic, it looks like there would not be a race condition because each transaction would only have one thread bumping the gas. 

I still think that this function would be useful for handling error if the coin does not exist. Instead of the caller having to define the error everytime the error is being called.